source/remoteClient/server.py

```diff
@@ -0,0 +1,474 @@
+	Creates IPv4 and IPv6 listening sockets using SSL/TLS encryption.
+	Uses select() for non-blocking I/O and monitors connection health with periodic pings
+	(sent every PING_TIME seconds, no response expected).
+
+	Clients must authenticate by providing the correct channel password in their join message
+	before they can exchange messages. Both IPv4 and IPv6 clients share the same channel
+	and can interact with each other transparently.
+	"""
+
+	PING_TIME: int = 300
+
+	def __init__(
+		self,
+		port: int,
+		password: str,
+		bind_host: str = "",
+		bind_host6: str = "[::]:",
+		cert_dir: Optional[Path] = None,
+	):
+		self.port = port
+		self.password = password
+		self.cert_manager = RemoteCertificateManager(cert_dir)
+		self.cert_manager.ensureValidCertExists()
+
+		# Initialize other server components
+		self.serializer = JSONSerializer()
+		self.clients: Dict[socket.socket, Client] = {}
+		self.clientSockets: List[socket.socket] = []
+		self._running = False
+		self.lastPingTime = 0
+
+		# Create server sockets
+		self.serverSocket = self.createServerSocket(
+			socket.AF_INET,
+			socket.SOCK_STREAM,
+			bind_addr=(bind_host, self.port),
+		)
+		self.serverSocket6 = self.createServerSocket(
+			socket.AF_INET6,
+			socket.SOCK_STREAM,
+			bind_addr=(bind_host6, self.port),
+		)
+
+	def createServerSocket(self, family: int, type: int, bind_addr: Tuple[str, int]) -> ssl.SSLSocket:
+		"""Creates an SSL wrapped socket using the certificate."""
+		serverSocket = socket.socket(family, type)
+		ssl_context = self.cert_manager.createSSLContext()
+		serverSocket = ssl_context.wrap_socket(serverSocket, server_side=True)
+		serverSocket.bind(bind_addr)
+		serverSocket.listen(5)
+		return serverSocket
+
+	def run(self) -> None:
+		"""Main server loop that handles client connections and message routing."""
+		log.info(f"Starting NVDA Remote relay server on ports {self.port} (IPv4) " f"and {self.port} (IPv6)")
+		self._running = True
+		self.lastPingTime = time.time()
+		while self._running:
+			r, w, e = select(
+				self.clientSockets + [self.serverSocket, self.serverSocket6],
+				[],
+				self.clientSockets,
+				60,
+			)
+			if not self._running:
+				break
+			for sock in r:
+				if sock is self.serverSocket or sock is self.serverSocket6:
+					self.acceptNewConnection(sock)
+					continue
+				self.clients[sock].handleData()
+			if time.time() - self.lastPingTime >= self.PING_TIME:
+				for client in self.clients.values():
+					if client.authenticated:
+						client.send(type=RemoteMessageType.PINGping)
+				self.lastPingTime = time.time()
+
+	def acceptNewConnection(self, sock: ssl.SSLSocket) -> None:
+		"""Accept and set up a new client connection."""
+		try:
+			clientSock, addr = sock.accept()
+			log.info(f"New client connection from {addr}")
+		except (ssl.SSLError, socket.error, OSError):
+			log.error("Error accepting connection", exc_info=True)
+			return
+		clientSock.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
```

Comment:
Can you add an explanatory comment here?
```suggestion
		# Disable Nagle's algorithm so that packets are always sent immediately.
		clientSock.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
```